code/core/src/core-server/build-dev.ts

```diff
@@ -29,18 +31,33 @@ import { warnWhenUsingArgTypesRegex } from './utils/warnWhenUsingArgTypesRegex';
 import { buildOrThrow } from './utils/build-or-throw';
 
 export async function buildDevStandalone(
-  options: CLIOptions & LoadOptions & BuilderOptions
+  options: CLIOptions &
+    LoadOptions &
+    BuilderOptions & {
+      storybookVersion?: string;
+      previewConfigPath?: string;
+    }
 ): Promise<{ port: number; address: string; networkAddress: string }> {
   const { packageJson, versionUpdates } = options;
-  invariant(
-    packageJson.version !== undefined,
-    `Expected package.json#version to be defined in the "${packageJson.name}" package}`
-  );
+  let { storybookVersion, previewConfigPath } = options;
+  const configDir = resolve(options.configDir);
+  if (packageJson) {
+    invariant(
+      packageJson.version !== undefined,
+      `Expected package.json#version to be defined in the "${packageJson.name}" package}`
+    );
+    storybookVersion = packageJson.version;
+    previewConfigPath = getConfigInfo(packageJson, configDir).previewConfig;
```

Comment:
```suggestion
    previewConfigPath = getConfigInfo(packageJson, configDir).previewConfig ?? undefined;
```
Something like this could fix the problem if `previewConfig` is `null`.